### PR TITLE
remove dependency from magento setup exception

### PR DESCRIPTION
**Pull request by rvitaliy**
**Description**
remove dependency from `Magento\Setup\Exception` it's not a vendor exception.
plus I have removed some try catches that are no longer needed.

**Tested scenarios**
no tests needed

**Fixed issue**:  none

### CODE REVIEW

File: Gateway/Request/ApplePayAuthorizationDataBuilder.php [hunk 1]
```diff
@@ -65,13 +64,7 @@ public function build(array $buildSubject)
...
         if ($token) {
             $parsedToken = json_decode($token);
-            $paymentData = $parsedToken->token->paymentData;
-            try {
-                $paymentData = base64_encode(json_encode($paymentData));
-                $requestBody['paymentMethod']['applepay.token'] = $paymentData;
-            } catch (\Exception $exception) {
-                $this->_adyenLogger->addAdyenDebug("exception: " . $exception->getMessage());
-            }
+            $requestBody['paymentMethod']['applepay.token'] = base64_encode(json_encode($parsedToken->token->paymentData));;
```
Comment (cyattilakiss) [inline Gateway/Request/ApplePayAuthorizationDataBuilder.php:67]:
```suggestion
            $requestBody['paymentMethod']['applepay.token'] = base64_encode(json_encode($parsedToken->token->paymentData));
```

**Comment by cyattilakiss** [comment]
Hi @rvitaliy 
Thank you for the PR, we are going to review it soon :)
Best,
Attila
Adyen

**Comment by acampos1916** [comment]
Thanks @rvitaliy!

@cyattilakiss I see we get a code smell about the Exception being of a generic type, do you think it should be an AdyenException instead?

**Comment by cyattilakiss** [comment]
> Thanks @rvitaliy!
> 
> @cyattilakiss I see we get a code smell about the Exception being of a generic type, do you think it should be an AdyenException instead?

yes that sounds good!

**Comment by acampos1916** [comment]
Hey @rvitaliy, are you able to switch the Exception type thrown to AdyenException to fix the generic type code smell?

**Review by cyattilakiss — APPROVED** [review]

**Review by acampos1916 — APPROVED** [review]

**Pull request merged**